docs/index.rst

```diff
@@ -127,6 +127,13 @@ Contents
    cheatsheet.rst
    grammar.rst
 
+.. toctree::
+   :maxdepth: 2
+   :caption: Compiler
+
+   using-the-compiler.rst
+   analysing-compilation-output.rst
```

Comment:
Added this new big section.

Reply:
I think some other pages also fall into this section, but let's leave it like that for now.